Assets/Plugins/Editor/Uplift/MenuItems.cs

```diff
@@ -70,39 +71,50 @@ private static void EditUpfile()
-                string string_latest = string.IsNullOrEmpty(latest) ? "No version available in any repository" : "Latest version is " + latest;
-                if (installed)
+                UpliftManager.DependencyState[] states = UpliftManager.Instance().GetDependenciesState(false);
+
+                foreach(UpliftManager.DependencyState state in states)
                 {
-                    string installed_version = upbring.GetInstalledPackage(name).Version;
-                    if (installed_version != latest)
+                    string message = string.Format("Package {0} ({1}) ", state.definition.Name, state.definition.Version);
+                    if(state.installed != null)
                     {
-                        Debug.Log(string.Format("Package {0} is outdated: installed version is {1}, latest is {2}", name, installed_version, string_latest));
+                        if(state.installed.Version != state.bestMatch.Package.PackageVersion)
+                            message += string.Format(
+                                "is outdated. Best available version is {0} (from {1})",
+                                state.bestMatch.Package.PackageVersion,
+                                state.bestMatch.Repository.ToString()
+                            );
+                        else
+                            message += "is up to date.";
+                        
+                        if(!state.definition.Requirement.IsMetBy(state.installed.Version))
+                            message += "\nWarning: the package currently installed does not match your requirements";
+                    
                     }
                     else
-                    {
-                        Debug.Log(string.Format("Package {0} is up-to-date ({1})", name, installed_version));
-                    }
-                }
-                else
-                {
-                    Debug.Log(string.Format("Package {0} is not installed ({1})", name, string_latest));
+                        message += string.Format(
+                            "is not installed. Best available version is {0} (from {1})",
+                            state.bestMatch.Package.PackageVersion,
+                            state.bestMatch.Repository.ToString()
+                        );
+
+                    if(state.latest.Package.PackageVersion != state.bestMatch.Package.PackageVersion)
+                        message += string.Format(
+                            "\nNote: there is a more recent version of the package ({0} from {1}), but it doesn't match your requirement",
+                            state.latest.Package.PackageVersion,
+                            state.bestMatch.Repository.ToString()
+                        );
+
+                    if(state.transitive)
+                        message = "`--> " + string.Join("    \n", message.Split('\n'));
```

Comment:
was the ` character expected?

Reply:
Yes, so this looks a bit like an arrow going down. I think it looks good, but we can remove it if you want

Reply:
```
‎________$$$$
_______$$__$
_______$___$$
_______$___$$
_______$$___$$
________$____$$
________$$____$$$
_________$$_____$$
_________$$______$$
__________$_______$$
____$$$$$$$________$$
__$$$_______________$$$$$$
_$$____$$$$____________$$$
_$___$$$__$$$____________$$
_$$________$$$____________$
__$$____$$$$$$____________$
__$$$$$$$____$$___________$
__$$_______$$$$___________$
___$$$$$$$$$__$$_________$$
____$________$$$$_____$$$$
____$$____$$$$$$____$$$$$$
_____$$$$$$____$$__$$
_______$_____$$$_$$$
________$$$$$$$$$$
```

Reply:
This is a bit confusing though, do you approve, or do you think we should indeed remove it

Reply:
keep it